packages/analytics-js/src/components/eventManager/EventManager.ts

```diff
@@ -1,18 +1,263 @@
+   * @param category Page's category
+   * @param name Page name
+   * @param properties Page properties
+   * @param options API options
+   * @param callback Callback function
+   */
+  private processPageEvent(
+    category?: Nullable<string>,
+    name?: Nullable<string>,
+    properties?: Nullable<ApiObject>,
+    options?: Nullable<ApiOptions>,
+    callback?: Nullable<ApiCallback>,
+  ): void {
+    // TODO: Handle adblocker request
+
+    let props = R.clone(properties) || {};
+    const opts = R.clone(options);
+
+    if (props) {
+      props.name = name;
+      props.category = category;
+    }
+
+    props = getUpdatedPageProperties(props, opts);
+
+    const pageEvent: Partial<RudderEvent> = {
+      properties: props,
+      name,
+      category,
+      type: RudderEventType.PAGE,
+      ...getCommonEventData(props),
+    };
+
+    this.processEvent(pageEvent as RudderEvent, opts, callback);
+  }
+
+  /**
+   * Generate a 'track' event based on the user-input fields
+   * @param event The event name
+   * @param properties Event properties
+   * @param options API options
+   * @param callback Callback function
+   */
+  private processTrackEvent(
+    event: string,
+    properties?: Nullable<ApiObject>,
+    options?: Nullable<ApiOptions>,
+    callback?: Nullable<ApiCallback>,
+  ): void {
+    const props = R.clone(properties);
+    const opts = R.clone(options);
+    const trackEvent: Partial<RudderEvent> = {
+      properties: props,
+      event,
+      type: RudderEventType.TRACK,
+      ...getCommonEventData(),
+    };
+
+    this.processEvent(trackEvent as RudderEvent, opts, callback);
+  }
+
+  /**
+   * Generate an 'identify' event based on the user-input fields
+   * @param userId New user ID
+   * @param traits User traits
+   * @param options API options
+   * @param callback Callback function
+   */
+  private processIdentifyEvent(
+    userId?: Nullable<string>,
+    traits?: Nullable<ApiObject>,
+    options?: Nullable<ApiOptions>,
+    callback?: Nullable<ApiCallback>,
+  ): void {
+    const cTraits = R.clone(traits);
+    const opts = R.clone(options);
+
+    batch(() => {
+      sessionState.rl_user_id.value = tryStringify(userId);
+      sessionState.rl_trait.value = { ...sessionState.rl_trait.value, ...cTraits };
+    });
+
+    const identifyEvent: Partial<RudderEvent> = {
+      type: RudderEventType.IDENTIFY,
+      ...getCommonEventData(),
+    };
+
+    this.processEvent(identifyEvent as RudderEvent, opts, callback);
+  }
+
+  /**
+   * Generate an 'alias' event based on the user-input fields
+   * @param to New user ID
+   * @param from Old user ID
+   * @param options API options
+   * @param callback Callback function
+   */
+  private processAliasEvent(
+    to: string,
+    from?: Nullable<string>,
+    options?: Nullable<ApiOptions>,
+    callback?: Nullable<ApiCallback>,
+  ): void {
+    const opts = R.clone(options);
+
+    sessionState.rl_user_id.value = tryStringify(to);
```

Comment:
Line 169 should occur after 176 as you are updating the sessionState value here.

Reply:
Addressed.